app/components/Sidebar/components/CollectionLink.tsx

```diff
@@ -197,19 +198,28 @@ function CollectionLink({
   const isDraggingAnyCollection =
     isDraggingAnotherCollection || isCollectionDragging;
 
+  const expandedStateBeforeDragging = React.useRef<boolean | null>(null);
```

Comment:
This approach using refs works but I don't prefer it stylistically. It overloads the meaning of `expanded` from "the user chose this to be expanded" to "the user may have chosen this, or it may be for other reasons". Better to make this explicit and not layer it on with refs.

Note that we already have all of the information with `isDraggingAnyCollection` and `expanded`

Try something like this pseudocode
```
expanded = useState(...)
isDraggingAnyCollection = ...

displayExpanded = expanded & !isDraggingAnyCollection
```